### PR TITLE
[SeiDB] Pt 2 State Store Unit Tests

### PR DESCRIPTION
**Pull request by Kbhat1**
- Adds a benchmark test suit to state store
- Adds more unit tests to state store

### CODE REVIEW

File: ss/test/storage_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
+//go:build rocksdbBackend
+// +build rocksdbBackend
```
Comment (yzang2019) [inline ss/test/storage_bench_test.go:2]: can we remove the build tag?

File: ss/test/storage_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/cosmos/iavl"
+	"github.com/sei-protocol/sei-db/proto"
+	"github.com/sei-protocol/sei-db/ss/pebbledb"
+	"github.com/sei-protocol/sei-db/ss/rocksdb"
+	"github.com/sei-protocol/sei-db/ss/sqlite"
+	"github.com/sei-protocol/sei-db/ss/types"
+)
+
+var (
+	backends = map[string]func(dataDir string) (types.StateStore, error){
```
Comment (yzang2019) [inline ss/test/storage_bench_test.go:24]: The test doesn't really need to be aware of different backends right? Can we just use the the SS interface?

File: ss/test/storage_test_suite.go [hunk 3]
```diff
@@ -475,3 +490,243 @@ func (s *StorageTestSuite) TestDatabasePrune() {
...
+	// write a key at three different versions 1, 100 and 200
+	cs := &iavl.ChangeSet{
+		Pairs: []*iavl.KVPair{{Key: key, Value: []byte("value001")}},
+	}
+	ncs := &proto.NamedChangeSet{
+		Name:      storeKey1,
+		Changeset: *cs,
+	}
+	s.Require().NoError(db.ApplyChangeset(1, ncs))
+
```
Comment (yzang2019) [inline ss/test/storage_test_suite.go:514]: Can we abstract this into a helper function?

File: ss/test/storage_test_suite.go [hunk 4]
```diff
@@ -475,3 +490,243 @@ func (s *StorageTestSuite) TestDatabasePrune() {
...
+				s.Require().NoError(err)
+				if v > int64(i) {
+					s.Require().NoError(db.Prune(v - 1))
+					break
+				}
+			}
+		}
+	}()
+
+	// start the goroutines
```
Comment (yzang2019) [inline ss/test/storage_test_suite.go:718]: you mean wait for the goroutine?

File: ss/test/storage_test_suite.go [hunk 5]
```diff
@@ -475,3 +490,243 @@ func (s *StorageTestSuite) TestDatabasePrune() {
...
+			val := fmt.Sprintf("val-%d-%03d", i, j)
+
+			v, err := db.Get(storeKey1, version, []byte(key))
+			s.Require().NoError(err)
+			s.Require().Equal([]byte(val), v)
+		}
+	}
+}
+
+func (s *StorageTestSuite) TestParallelWriteAndPruning() {
```
Comment (yzang2019) [inline ss/test/storage_test_suite.go:667]: Does this unit test catch the issue where we hit the issue with clone?

**Review by yzang2019 — APPROVED** [review]

File: ss/test/storage_test_suite.go [hunk 6]
```diff
@@ -475,3 +475,67 @@
...
+		Pairs: []*iavl.KVPair{{Key: key, Value: []byte("value003")}},
+	}
+	ncs.Changeset = *cs
+	s.Require().NoError(db.ApplyChangeset(200, ncs))
+
+	// prune version 50
+	s.Require().NoError(db.Prune(50))
+
+	// ensure queries for versions 50 and older return nil
+	bz, err := db.Get(storeKey1, 49, key)
```
[the remark this answers is not in the document]
Comment (mishkajul) [reply ss/test/storage_test_suite.go:531]: B

File: ss/test/storage_bench_test.go (hunk 1 above)
Comment (mishkajul) [reply ss/test/storage_bench_test.go:2]: []()

Comment (Kbhat1) [reply ss/test/storage_bench_test.go]: Its now updated without build tags and not backend specific

**Pull request merged**